src/pages/api/admin/bulk-data-import.ts

```diff
@@ -0,0 +1,73 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+import type { NextApiRequest, NextApiResponse } from "next";
+import { ReasonPhrases, StatusCodes } from "http-status-codes";
+import { ErrorWithCause } from "pony-cause";
+import { services } from "../../../services/ServiceLocator";
+import { BulkDataImportStatus } from "../../../services/AppModel";
+import { serverLogError } from "../log";
+
+const IMPORT = "import";
+
+interface ValidRequest {
+  action: typeof IMPORT;
+}
+
+function validateMethod(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== "PUT") {
```

Comment:
I know you hate writing this out like this, so thanks for taking the simplest approach rather than looking for a generic declarative solution.